state-chain/runtime/src/chainflip/solana_elections.rs

```diff
@@ -227,11 +234,22 @@ impl
 		alt_identifier: SolanaAltWitnessingIdentifier,
 		alts: AltConsensusResult<Vec<SolAddressLookupTableAccount>>,
 	) {
-		Environment::add_sol_ccm_swap_alts(alt_identifier.swap_request_id, alts);
-	}
-
-	fn should_expire_election(alt_identifier: SolanaAltWitnessingIdentifier) -> bool {
-		crate::System::block_number() >= alt_identifier.election_expiry_block_number
+		// use set_unsynchronised_state_map to write result onto the state map
+		match alts {
+			AltConsensusResult::ValidConsensusAlts(res) => res.into_iter().for_each(|r| {
+				SolanaAltWitnessingElectoralAccess::set_unsynchronised_state_map(
+					r.key.into(),
+					Some(AltConsensusResult::ValidConsensusAlts(r)),
+				)
+			}),
+			AltConsensusResult::AltsInvalidNoConsensus =>
+				alt_identifier.alt_addresses.into_iter().for_each(|alt| {
+					SolanaAltWitnessingElectoralAccess::set_unsynchronised_state_map(
```

Comment:
Similar comment here, ideally we don't want to have to deal with pallet internals here. We should replace the hook with a storage write, and then provide a lookup function to 'pull' the data.